### PR TITLE
macOS support

### PR DESCRIPTION
**Pull request by garakmon**
These changes are for people using pokeemerald on macOS 10.12 or higher.

### CODE REVIEW

**Comment by garakmon** [comment]
Hi :) I don't thing any of the changes I made here will affect building of this ROM on other systems so hopefully the build passes and the changes are approvable.

**Comment by Diegoisawesome** [comment]
Pending multiplatform tests.

File: Makefile [hunk 1]
```diff
@@ -25,13 +25,19 @@ ASFLAGS := -mcpu=arm7tdmi
 CC1             := tools/agbcc/bin/agbcc
 override CFLAGS += -mthumb-interwork -Wimplicit -Wparentheses -Werror -O2 -fhex-asm
 
-CPPFLAGS := -I tools/agbcc/include -iquote include -nostdinc -undef
+CPPFLAGS := -I tools/agbcc/include -I tools/agbcc -iquote include -nostdinc -undef
```
Comment (garakmon) [inline Makefile]: this is necessary unless agbcc PR#22 is merged (which seems unlikely)

File: Makefile [hunk 2]
```diff
@@ -25,13 +25,19 @@ ASFLAGS := -mcpu=arm7tdmi
...
 
 LDFLAGS = -Map ../../$(MAP)
 
 LIB := -L ../../tools/agbcc/lib -lgcc -lc
 
-SHA1 := sha1sum -c
+ifeq ($(OS),Windows_NT)
+	SHA1 := sha1sum -c
+else ifeq ($(shell uname -s),Darwin)
```
Comment (Diegoisawesome) [inline Makefile]: Couldn't this just be a check or Darwin and an else for both NT and others?

**Pull request merged**